### PR TITLE
feat: support rules_js 1.6.5 with improved version mirroring

### PR DESCRIPTION
**Pull request by gregmagolan**
Depends on https://github.com/aspect-build/rules_js/pull/576 landing first.

Introduces new npm tool mirror that vendors only the generated pnpm-lock.yaml file and not the generated defs.bzl and repositories.bzl from npm_translate_lock. The old way was no longer trivial with the introduction of aliases in the npm_translate_lock repository for bzlmod support.

The new way comes at the cost of an additional load & function call in a user's WORKSPACE, but it is simpler and more portable as generated starlark from npm_translate_lock is no longer baked but rather dynamic based on the rules_js version being used.

### CODE REVIEW

File: jasmine/dependencies.bzl
Comment (jbedard) [inline jasmine/dependencies.bzl:24]: Was there any reason to have a mirror before?

Comment (gregmagolan) [reply jasmine/dependencies.bzl:24]: The bazel build mirror urls is just copy pasted around. githbu.com is sufficient and adding the bazel mirror as a fallback doesn't add much value imo.

**Review by thesayyn — APPROVED** [review]
LGTM. I really like that we don't have to check in the generated bzl file.

File: jasmine/dependencies.bzl
Comment (jbedard) [inline jasmine/dependencies.bzl:33]: Lets wait for the release before merging though?

Comment (gregmagolan) [reply jasmine/dependencies.bzl:33]: Yup

**Review by jbedard — APPROVED** [review]

**Pull request merged**
